### PR TITLE
Fixed the infinite loop problem on #18

### PR DESCRIPTION
**Pull request by sekikn**
Specifying maven as the build tool on ant project (or vice versa) have caused an infinite loop. See #18.

### CODE REVIEW

File: precommit/main/test-patch.sh [hunk 1]
```diff
@@ -1018,11 +1025,19 @@ function find_changed_modules
...
+          exit 1
+        fi
+        pomdirs="${pomdirs} ${pomdir}"
       ;;
       ant)
         #shellcheck disable=SC2086
-        pomdirs="${pomdirs} $(find_buildxml_dir ${file})"
+        pomdir=$(find_buildxml_dir ${file})
+        if [[ -z ${pomdir} ]]; then
+          exit 1
```
Comment (aw-altiscale) [inline precommit/main/test-patch.sh]: same thing.

File: precommit/main/test-patch.sh [hunk 2]
```diff
@@ -1018,11 +1025,19 @@ function find_changed_modules
     case ${BUILDTOOL} in
       maven)
         #shellcheck disable=SC2086
-        pomdirs="${pomdirs} $(find_pomxml_dir ${file})"
+        pomdir=$(find_pomxml_dir ${file})
+        if [[ -z ${pomdir} ]]; then
+          exit 1
```
Comment (aw-altiscale) [inline precommit/main/test-patch.sh]: can't be exit.  needs to be 

```bash
        output_to_console 1
        output_to_bugsystem 1
        cleanup_and_exit 1
```

**Comment by sekikn** [comment]
Thanks, Allen. It looks fine.

```
[sekikn@localhost pre-yetus]$ precommit/main/test-patch.sh --basedir=../hadoop --personality=precommit/main/personality/hadoop.sh --build-tool=ant HADOOP-12070


============================================================================
============================================================================
                        Bootstrapping test harness
============================================================================
============================================================================


/tmp/test-patch-yetus/17480 has been created
Running in developer mode
 patch is being downloaded at Wed Jun 17 06:41:43 UTC 2015 from
http://issues.apache.org/jira/secure/attachment/12738174/HADOOP-12070.001.patch


============================================================================
============================================================================
                        Confirming git environment
============================================================================
============================================================================


Testing HADOOP-12070 patch on trunk.
Checking patch hadoop-common-project/hadoop-common/src/main/bin/hadoop.cmd...
ERROR: build.xml is not found. Make sure the target is a Ant-based project.


-1 overall

 _____     _ _                _ 
|  ___|_ _(_) |_   _ _ __ ___| |
| |_ / _` | | | | | | '__/ _ \ |
|  _| (_| | | | |_| | | |  __/_|
|_|  \__,_|_|_|\__,_|_|  \___(_)
                                


| Vote |    Subsystem |  Runtime   | Comment
============================================================================


|| Subsystem || Report/Notes ||
============================================================================
| Java | 1.7.0_79 |
| uname | Linux localhost.localdomain 2.6.32-504.16.2.el6.x86_64 #1 SMP Wed Apr 22 06:48:29 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | ant |
| Personality | precommit/main/personality/hadoop.sh |
| Patch URL | http://issues.apache.org/jira/secure/attachment/12738174/HADOOP-12070.001.patch |
| git revision | trunk / 5dbc8c9 |
| Optional Tests | asflicense |


============================================================================
============================================================================
                              Finished build.
============================================================================
============================================================================


[sekikn@localhost pre-yetus]$ precommit/main/test-patch.sh --basedir=../pig PIG-4598


============================================================================
============================================================================
                        Bootstrapping test harness
============================================================================
============================================================================


/tmp/test-patch-yetus/17848 has been created
Running in developer mode
 patch is being downloaded at Wed Jun 17 06:47:25 UTC 2015 from
http://issues.apache.org/jira/secure/attachment/12739032/PIG-4598.patch


============================================================================
============================================================================
                        Confirming git environment
============================================================================
============================================================================


WARNING: Current git branch is trunk but patch is built for master.
WARNING: Continuing anyway...
Testing patch on trunk.
Checking patch src/org/apache/pig/PigConstants.java...
Checking patch src/org/apache/pig/newplan/logical/optimizer/LogicalPlanOptimizer.java...
ERROR: pom.xml is not found. Make sure the target is a Maven-based project.


-1 overall

 _____     _ _                _ 
|  ___|_ _(_) |_   _ _ __ ___| |
| |_ / _` | | | | | | '__/ _ \ |
|  _| (_| | | | |_| | | |  __/_|
|_|  \__,_|_|_|\__,_|_|  \___(_)
                                


| Vote |    Subsystem |  Runtime   | Comment
============================================================================


|| Subsystem || Report/Notes ||
============================================================================
| Java | 1.7.0_79 |
| uname | Linux localhost.localdomain 2.6.32-504.16.2.el6.x86_64 #1 SMP Wed Apr 22 06:48:29 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Patch URL | http://issues.apache.org/jira/secure/attachment/12739032/PIG-4598.patch |
| git revision | trunk / bcbb68e |
| Optional Tests | asflicense javac javadoc mvninstall unit findbugs checkstyle |


============================================================================
============================================================================
                              Finished build.
============================================================================
============================================================================


[sekikn@localhost pre-yetus]$ 
```

**Comment by dierobotsdie** [comment]
BTW, this code gets run twice now (vs. what's in hadoop).  So if a patch removes the pom.xml/build.xml, we should be able to detect that now. :)  So let's add a -1 vote (subsystem build? I dunno.) and we should be good to go.

**Comment by aw-altiscale** [comment]
Basically, this code gets executed twice now so that we can detect if modules have been added/removed during the patch. (thus why we can't just exit) So we should now be able to detect if a patch changes the build system in such a way that the pom.xml/build.xml files get removed in a bad way.  So I think we just need a vote now (subsystem build or test or something?) and we should be good to go.

**Comment by aw-altiscale** [comment]
Let's cover the vote issue later. I want to get this merged in now. :)

**Pull request merged**